libs/beacon/src/beacon_manager.cpp

```diff
@@ -17,167 +17,617 @@
 //------------------------------------------------------------------------------
 
 #include "beacon/beacon_manager.hpp"
+#include "crypto/ecdsa.hpp"
 
 namespace fetch {
 namespace dkg {
 
-void BeaconManager::Reset(uint64_t cabinet_size, uint32_t threshold)
+bn::G2 BeaconManager::zeroG2_;
```

Comment:
These are statics right? I think they can not be static in beacon_manager.hpp and just construct them implicitly